tesseract_collision/include/tesseract_collision/contact_checker_common.h

```diff
@@ -124,6 +129,233 @@ inline ContactResult* processResult(ContactDistanceData& cdata,
 }
+
+/**
+ * @brief Create a convex hull from vertices using Bullet Convex Hull Computer
+ * @param (Output) vertices A vector of vertices
+ * @param (Output) faces The first values indicates the number of vertices that define the face followed by the vertice index
+ * @param (input) input A vector of point to create a convex hull from
+ * @param (input) shrink If positive, the convex hull is shrunken by that amount (each face is moved by "shrink" length
+ *                units towards the center along its normal).
+ * @param (input) shrinkClamp If positive, "shrink" is clamped to not exceed "shrinkClamp * innerRadius", where
+ *                "innerRadius" is the minimum distance of a face to the center of the convex hull.
+ * @return The number of faces. If less than zero an error occured when trying to create the convex hull
+ */
+inline int createConvexHull(VectorVector3d& vertices, std::vector<int>& faces, const VectorVector3d& input, double shrink = -1, double shrinkClamp = -1)
+{
+  vertices.clear();
+  faces.clear();
+
+  btConvexHullComputer conv;
+  btAlignedObjectArray<btVector3> points;
+  points.reserve(input.size());
+  for (const auto& v : input)
+  {
+    points.push_back(btVector3(v[0], v[1], v[2]));
+  }
+
+  btScalar val = conv.compute(&points[0].getX(), sizeof(btVector3), points.size(), shrink, shrinkClamp);
+  if (val < 0)
+  {
+    ROS_ERROR("Failed to create convex hull");
+    return -1;
+  }
+
+  int num_verts = conv.vertices.size();
+  vertices.reserve(num_verts);
+  for (int i = 0; i < num_verts; i++)
+  {
+    btVector3& v = conv.vertices[i];
+    vertices.push_back(Eigen::Vector3d(v.getX(), v.getY(), v.getZ()));
+  }
+
+  int num_faces = conv.faces.size();
+  faces.reserve(3 * num_faces);
+  for (int i = 0; i < num_faces; i++)
+  {
+    std::vector<int> face;
+    face.reserve(3);
+
+    const btConvexHullComputer::Edge* sourceEdge = &(conv.edges[conv.faces[i]]);
+    int a = sourceEdge->getSourceVertex();
+    face.push_back(a);
+
+    int b = sourceEdge->getTargetVertex();
+    face.push_back(b);
+
+    const btConvexHullComputer::Edge* edge = sourceEdge->getNextEdgeOfFace();
+    int c = edge->getTargetVertex();
+    face.push_back(c);
+
+    edge = edge->getNextEdgeOfFace();
+    c = edge->getTargetVertex();
+    while (c != a)
+    {
+      face.push_back(c);
+
+      edge = edge->getNextEdgeOfFace();
+      c = edge->getTargetVertex();
+    }
+    faces.push_back(face.size());
+    faces.insert(faces.end(), face.begin(), face.end());
+  }
+
+  return num_faces;
+}
+
+/**
+ * @brief Write a simple ply file given vertices and faces
+ * @param path The file path
+ * @param vertices A vector of vertices
+ * @param faces The first values indicates the number of vertices that define the face followed by the vertice index
+ * @param num_faces The number of faces
+ */
+inline void writeSimplePlyFile(const std::string& path, const VectorVector3d vertices, const std::vector<int> &faces, int num_faces)
```

Comment:
`const VectorVector3d vertices` is not a reference. Is this intended?

Reply:
Nope should be a reference